streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```
@@ -293,6 +302,168 @@ public boolean isValidTransition(final State newState) {
+            final boolean allInternalTopicsExist = validationResult.missingTopics().isEmpty();
+            final boolean missingSourceTopics = !Collections.disjoint(validationResult.missingTopics(), allSourceTopics);
+
+            if (internalTopicsMisconfigured) {
+                throw new MisconfiguredInternalTopicException("Misconfigured Internal Topics: " + validationResult.misconfigurationsForTopics());
+            }
+            if (missingSourceTopics) {
+                allSourceTopics.retainAll(validationResult.missingTopics());
+                throw new MissingSourceTopicException("Missing source topics: " + allSourceTopics);
+            }
+            if (noInternalTopicsExist) {
+                internalTopicManager.setup(allInternalTopics);
+            } else if (allInternalTopicsExist) {
+                throw new InternalTopicsAlreadySetupException("All internal topics have already been setup");
+            } else {
+                if (initParameters.setupInternalTopicsIfIncompleteEnabled()) {
+                    final Map<String, InternalTopicConfig> topicsToCreate = new HashMap<>();
+                    for (final String missingTopic : validationResult.missingTopics()) {
+                        topicsToCreate.put(missingTopic, allInternalTopics.get(missingTopic));
+                    }
+                    internalTopicManager.makeReady(topicsToCreate);
+                } else {
+                    throw new MissingInternalTopicsException("Missing Internal Topics: ", new ArrayList<>(validationResult.missingTopics()));
+                }
+            }
+        } catch (final TimeoutException timeoutException) {
+            throw new TimeoutException(timeoutException.getMessage(), timeoutException);
+        } catch (final StreamsException streamsException) {
+            throw new StreamsException(streamsException.getMessage(), streamsException);
+        }
```

Not sure why we want to catch and rethrow? Seems we don't add any value doing this, but we can remove the complete try-catch block and let any exception just bubble up unmodified?

---

You're right, this is bad style - I will update this